app/views/admin/pages/_form.html.erb

```diff
@@ -38,9 +38,36 @@
     </div>
     <div class="form-group">
       <%= form.label :is_top_level_path %>
-      <%= form.check_box :is_top_level_path %>
-      <p>(Determines if it is accessible by <code>/page-slug</code> vs <code>/page/page-slug</code>) Be careful! ⚠️</p>
+      <%= form.check_box :is_top_level_path, class: "crayons-checkbox" %>
+      <p>Determines if it is accessible by <code>/page-slug</code> vs <code>/page/page-slug</code> Be careful! ⚠️</p>
     </div>
+    <% if @page.errors.count > 0 %>
+      <div id="add-landing-page-link-modal-root">
+        <% @page.errors %>
+      </div>
+    <% end %>
+      <span data-controller="modal" data-modal-root-selector-value="#add-landing-page-link-modal-root" data-modal-content-selector-value="#add-landing-page-link-modal"
+            data-modal-title-value="There's another locked screen..." data-modal-size-value="default">
+        <div class="form-group">
+          <%= form.label :landing_page, "Use as 'Locked Screen'" %>
+          <%= form.check_box :landing_page, class: "crayons-checkbox", data: { action: @page.has_a_landing_page? ? "modal#toggleModal" : "" } %>
```

Comment:
Currently, this modal is triggered upon checking this checkbox _if_ there is already an existing landing page (there is another page that is `landing_page: true`). That said, this works a little _too_ well currently and should only be triggered on the condition that there is already a landing page set _and_ that the landing page that is set is not the current landing page--I've found that if you decide to uncheck this box on the landing page that _is_ set, that the modal triggers and it probably shouldn't. 

I can image a situation where an Admin decides that they don't want _any_ landing pages for their private forem anymore and upon unchecking this box on their current landing page, they unexpectedly trigger this modal--the expectation here is that the modal only appears when the checkbox is checked on a landing page other than the current landing page.

Reply:
If in the Admin controller you load both `@landing_page` and `@page` you can check:

`@landing_page.present? && @landing_page.id != @page.id`